crates/stackable-operator/src/builder/pod/probe.rs

```diff
@@ -0,0 +1,330 @@
+use std::num::TryFromIntError;
+
+use k8s_openapi::{
+    api::core::v1::{ExecAction, GRPCAction, HTTPGetAction, Probe, TCPSocketAction},
+    apimachinery::pkg::util::intstr::IntOrString,
+};
+use snafu::{ResultExt, Snafu, ensure};
+
+use crate::time::Duration;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display(
+        "The probe's {field:?} duration of {duration} is too long, as it's seconds doesn't fit into an i32"
```

Comment:
How does that appear? Does it look like human time?
Eg:
- `duration of 4,734e+8 is too long`
- `duration of 15 is too long`, or
- `duration of 15 years is too long`